bundles/org.openhab.binding.ecovacs/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,351 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="ecovacs"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+
+	<thing-type id="vacuum">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="ecovacsapi"/>
+		</supported-bridge-type-refs>
+
+		<label>Ecovacs Vacuum Cleaner</label>
+		<description>Represents an Ecovacs vacuum cleaner</description>
+
+		<channel-groups>
+			<channel-group id="actions" typeId="actions"/>
+			<channel-group id="status" typeId="status"/>
+			<channel-group id="last-clean" typeId="last-clean"/>
+			<channel-group id="total-stats" typeId="total-stats"/>
+			<channel-group id="consumables" typeId="consumables"/>
+			<channel-group id="settings" typeId="settings"/>
+		</channel-groups>
+
+		<config-description>
+			<parameter name="serialNumber" type="text" required="true">
+				<label>Device Serial Number</label>
+			</parameter>
+			<parameter name="refresh" type="integer" unit="min" min="1" required="false">
+				<label>Refresh Interval</label>
+				<advanced>true</advanced>
+				<description>Specifies the refresh interval in minutes.</description>
+				<unitLabel>Minutes</unitLabel>
+				<default>5</default>
+			</parameter>
+		</config-description>
+	</thing-type>
+
+	<channel-group-type id="status">
+		<label>Status</label>
+		<channels>
+			<channel id="state" typeId="state"/>
+			<channel id="battery" typeId="system.battery-level"/>
+			<channel id="current-cleaning-mode" typeId="current-cleaning-mode"/>
```

Comment:
The current naming convention for channel ids is camelCase:
https://www.openhab.org/docs/developer/guidelines.html#java-coding-style

I believe the same convention is preferred for channel type ids as well:
```suggestion
			<channel id="currentCleaningMode" typeId="currentCleaningMode"/>
```

this applies to all channel ids and channel type ids.

Reply:
Is this a hard requirement? Grepping through the repo, I see all kinds of styles (`alllowercase`, `camelCase`, `using-hyphens`, `snake_case`), some use one style for channel IDs and another for channel type IDs (bluetooth.am43, hdpowerview). This also applies to rather new bindings like mercedesme, which was merged last year and whose code is actually *newer* than this PR (#13044).

The reason for my question is that I think it's a good idea to keep the current style otherwise, to make sure the people who started using the binding from the marketplace during the last year don't have to reassign all of their items, which is (when done via UI) a rather tedious experience.

Reply:
Not enforcing this rule exactly leads to the problem you mention where we end up with a mix of different naming conventions.

I don't know when this camelCase convention was introduced, but bindings created before then still suffers from this problem because channels cannot be renamed without breaking, and consistency within the binding takes precedence over cross-binding consistency. Meaning that if an old binding has all channels in snake_case, then new channels should keep this convention as well.

Today I would guess that camelCase is the mostly adopted convention with using-hyphens as a close second position. Probably due to the fact that this convention is used for system channel types.

You caught me with #13044. I don't remember if I missed it or if there was some argument for doing that.

You do have a valid argument with the existing community userbase, but on the other hand **only** those users will be affected, whereas **all** users would be affected if the renaming would happen after merging into the official distribution. So it's a bad moment, but there will never be a better moment. All new bindings could in reality be pre-released for the community without following the naming convention, and this would then have to be preserved because of existing users. This way it would be impossible to ever reach a common convention.

Back to your question, is it a "hard" requirement? For sure it's a documented official requirement, but I guess even maintainers can be soft. 😉 I would like to hear other opinions though from @openhab/add-ons-maintainers.

Reply:
Gentle ping @openhab/add-ons-maintainers

Reply:
@maniac103 - our naming convention seems not as graved in stone as I thought, and needs some further discussions and decisions. Your choice is consistent with the system channel ids and also by the convention used by many other bindings. Therefore we can merge as-is, so I'll resolve this comment.